### PR TITLE
refactor: Clean up config handling

### PR DESCRIPTION
**Pull request by mattsjones**
This PR cleans up config management in `sku`. `builds.js` & `builds.ssr.js` have been removed in favour of a centralised `context`. This makes default config much cleaner and easier to reason with.

Also removed all default config from test-cases. `sku-config.js` values should only be ones that diverge from the defaults. 

BREAKING CHANGE: 
- Removed support for monorepos
- Update SSR port config (no longer support `port` as object)

Note: This PR is being merge into a new `v7-beta` branch, not master.

### CODE REVIEW

File: config/webpack/webpack.config.ssr.js [hunk 1]
```diff
@@ -1,294 +1,291 @@
...
-          console.log(
-            `WARNING: Environment variable "${key}" for build "${name}" is missing a value for the "${
-              args.env
-            }" environment`
-          );
-          process.exit(1);
-        }
+    if (typeof valueForEnv === 'undefined') {
+      console.log(
+        `WARNING: Environment variable "${key}" for build "${name}" is missing a value for the "${
```
Comment (michaeltaranto) [inline config/webpack/webpack.config.ssr.js]: Remove `name` from here as well to be consistent with the other webpack config.

File: config/webpack/webpack.config.ssr.js [hunk 2]
```diff
@@ -1,294 +1,291 @@
...
+  : [skuServerEntry];
 
-    // Define publicPath
-    const publicPath = isStartScript
-      ? `http://localhost:${port.client}/`
-      : paths.publicPath || '';
+// Define publicPath
+const publicPath = isStartScript
+  ? `http://localhost:${port.client}/`
+  : paths.publicPath || '';
```
Comment (michaeltaranto) [inline config/webpack/webpack.config.ssr.js]: Again, lets look to make this logic consistent with the other webpack config

File: lib/runPrettier.js [hunk 3]
```diff
@@ -35,7 +35,11 @@ const runPrettier = async ({ write, listDifferent }) => {
     // don't error if `.prettierignore` not found
   }
 
-  prettierArgs.push('**/*.{js,ts,tsx,md,less,css}');
+  if (pathsToCheck.length > 0) {
+    prettierArgs.push(...pathsToCheck);
+  } else {
+    prettierArgs.push('**/*.{js,ts,tsx,md,less,css}');
+  }
```
Comment (michaeltaranto) [inline lib/runPrettier.js]: To simplify how this reads, maybe look to default `pathsToCheck` as `['**/*.{js,ts,tsx,md,less,css}']`.

File: test/test-cases/ssr-hello-world/sku.config.js [hunk 4]
```diff
@@ -1,9 +1,5 @@
 module.exports = {
-  entry: {
-    client: 'src/client.js',
-    server: 'src/server.js'
-  },
-  target: 'dist',
-  publicPath: 'http://localhost:8000/',
-  port: { client: 8000, backend: 8001 }
+  publicPath: 'http://some.cdn/',
```
Comment (mattsjones) [inline test/test-cases/ssr-hello-world/sku.config.js]: Changed this to better distinguish between `start-ssr` & `build-ssr` snapshots.

**Pull request merged**